server/src/main/java/org/elasticsearch/index/fielddata/IndexFieldData.java

```diff
@@ -72,6 +76,12 @@
      */
     SortField sortField(@Nullable Object missingValue, MultiValueMode sortMode, Nested nested, boolean reverse);
 
+    default BucketedSort newBucketedSort(BigArrays bigArrays, @Nullable Object missingValue, MultiValueMode sortMode, Nested nested,
+            SortOrder sortOrder, DocValueFormat format) {
+        // TODO implement all the fields.
+        throw new IllegalArgumentException("Unsupported field type");
```

Comment:
Could we have a default implementation based on the `sortField` implementation? My understanding is that SortField provides anything we need, BucketedSort is just nice because it has the concept of buckets that SortField doesn't have?

Reply:
Are you thinking of an implementation that uses an array of `SortField`s?

Reply:
yes

Reply:
I looked at this and it turned out to be kind of fiddly to get right. Given that we only want to support numeric sorts on his agg anyway it was easier to just remove this `TODO` and either implement the sort of throw an exception saying that it doesn't work on this field type.